Orso.Arpa.Domain/UserDomain/Commands/LoginUser.cs

```diff
@@ -66,13 +72,20 @@ public async Task<string> Handle(Command request, CancellationToken cancellation
                     throw new ValidationException(new[] { new ValidationFailure(nameof(request.UsernameOrEmail), "Your email address is not confirmed. Please confirm your email address first") });
                 }
 
-                SignInResult result = await _signInManager.CheckPasswordSignInAsync(user, request.Password, true);
+                bool IsCookieSignInPossible = await _cookieSignIn.IsCookieSignInPossible(user, request.Password);
 
-                if (result.Succeeded)
+                if (IsCookieSignInPossible)
                 {
-                    return await _jwtGenerator.CreateTokensAsync(user, request.RemoteIpAddress, cancellationToken);
+                    await _jwtGenerator.CreateRefreshTokenAsync(user, request.RemoteIpAddress, cancellationToken);
+
+                    Task<Task> signInTask = _cookieSignIn.SignInUser(user);
+                    await signInTask;
+
+                    return signInTask.IsCompletedSuccessfully;
                 }
 
+                SignInResult result = await _signInManager.PasswordSignInAsync(user, request.Password, false, true);
```

Comment:
this call manages the locked out case. if the password is wrong, we don't call above `SignInUserAsync` but this one, then it locks user if a certain number of wrong attempts happened.

Reply:
u call "IsCookieSignInPossible" which already checks for the correct password and user lock out in line 77

Reply:
`IsCookieSignInPossible` does not count down if the password is wrong, it only checks. we need something that manages lock out case by counting the wrong attempts. .`SignInUserAsync` does not do this either

Reply:
ok got this point. but why don't you use SignInManager.PasswordSignInAsync instead of IsCookieSignInPossible?
And: could you possibly use SignInManager.SignInWithClaimsAsync instead of setting the claims by hand?

Reply:
using `PasswordSignInAsync` takes the sign in action, i need something that will only check the password and decides whether sign in gonna be successful or not. If the login gonna be successful, using `PasswordSignInAsync` gonna result with sign in without cookies
I have also tried using `SignInWithClaimsAsync` but it had some issues while setting cookies that i can't remember off the top of my head what was the issue